res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,869 @@
+            engine.setValue(group, "loop_end_position", -1); //removes current to fix hanging out
+            engine.setValue(group, "loop_out", 1); //and sets again
+            print("value is " + value);
+        }
+    }
+};
+
+DenonHC1000S.reloop = function(group, value, shift) {
+    var loopInPosition = engine.getValue(group, "loop_start_position");
+    var loopOutPosition = engine.getValue(group, "loop_end_position");
+    if (loopInPosition !== -1 && loopOutPosition !== -1) {
+        engine.setValue(group, "reloop_toggle", 1);
+    } else {
+        var samplesPerBeat = DenonHC1000S.samplesPerBeat(group);
+        loopInPosition = engine.getValue(group, "playposition") * engine.getValue(group, "track_samples");
+        if (shift) {
+            var loopInPositionInBeats = loopInPosition / samplesPerBeat;
+            if ((loopInPositionInBeats - Math.floor(loopInPositionInBeats)) > 0.5) {
+                loopInPosition = Math.floor(0.5 + loopInPositionInBeats) * samplesPerBeat;
+            } else {
+                loopInPosition = Math.floor(loopInPositionInBeats) * samplesPerBeat;
+            }
+        } else {
+            loopInPosition = engine.getValue(group, "playposition") * engine.getValue(group, "track_samples");
+        }
+        loopInPosition = Math.floor(0.5 + loopInPosition);
+        loopInPosition = Math.max(0, loopInPosition - loopInPosition % 2);
+        loopOutPosition = loopInPosition + Math.floor(0.5 + samplesPerBeat);
+        loopOutPosition = Math.max(0, loopOutPosition - loopOutPosition % 2);
+        if (loopInPosition + samplesPerBeat < engine.getValue(group, "track_samples")) {
+            engine.setValue(group, "loop_start_position", loopInPosition);
+            engine.setValue(group, "loop_end_position", loopOutPosition);
+            engine.setValue(group, "reloop_toggle", 1);
+        }
+    }
+};
+
+DenonHC1000S.resizeLoop = function(midino, control, value, status, group) {
+    var knobDelta = DenonHC1000S.getKnobDelta(value);
+    print(group + " loop_scale " + knobDelta);
+    var scaleFactor = 1;
+    if (!DenonHC1000S.shift) {
+        if (knobDelta > 0) { //make scale bigger (right turn)
+            if (!DenonHC1000S.resizeLoopButtonPressed) {
+                scaleFactor = (scaleFactor * scaleFactor + 1 / 2);
+                print("scale up by factor " + scaleFactor);
+                engine.setValue(group, "loop_scale", scaleFactor);
+            } else {
+                //move loop to right
+                DenonHC1000S.moveLoopRight(midino, control, value, status, group);
+                print("moving loop right");
+            }
+        } else { //make scale smaller (left turn)
+            if (!DenonHC1000S.resizeLoopButtonPressed) {
+                scaleFactor = (scaleFactor - scaleFactor * 1 / 2);
+                print("scale down by factor " + scaleFactor);
+                engine.setValue(group, "loop_scale", scaleFactor);
```

Comment:
This just halves the loop size. I assume you want to use [`loop_halve`](https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-loop_halve) instead.